libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/MapRouteLine.kt

```diff
@@ -706,7 +706,7 @@ internal class MapRouteLine(
         setPrimaryRoutesSource(routeData.featureCollection)
         updateRouteTrafficSegments(routeData)
         if (style.isFullyLoaded) {
-            val expression = getExpressionAtOffset(0f)
+            val expression = getExpressionAtOffset(vanishPointOffset)
```

Comment:
The new route mos-likely has a different length than the old one. Is it safe to use the old percentage traveled when redrawing the route? Isn't that going to result in a vanishing point rendered ahead/behind of the puck?

Reply:
In the two places this method gets called there are also calls to hideRouteLineAtOffset and hideCasingLineAtOffset and in those cases the vanishingPointOffset is passed as the parameter.  I don't think there's a case when the traffic line vanishing point would ever be different than the other two lines.  I added vanishPointOffset above to match those method calls.